src/stats.jl

```diff
@@ -246,18 +243,24 @@ function summarystats(
     append_chains::Bool = true,
     method::MCMCDiagnosticTools.AbstractESSMethod = ESSMethod(),
     maxlag = 250,
-    etype = :bm,
+    skip_missing=true,
     kwargs...
 )
     # Store everything.
-    funs = [mean∘cskip, std∘cskip, sem∘cskip, x -> MCMCDiagnosticTools.mcse(cskip(x); method=etype, kwargs...)]
-    func_names = [:mean, :std, :naive_se, :mcse]
+    funs = [
+        mean, 
+        std,
+        MCMCDiagnosticTools.mcse,
+    ]
+    func_names = [:mean, :std, :mcse]
```

Comment:
These changes seem unrelated and undesired. Can you revert them?

Reply:
This is related, `cskip` doesn't work with weights, and `mcse` needs to be passed like this to make sure it works either with or without weights.

`naive_se` was removed because there's not really a way to define the naive standard error for a weighted chain (since weighted chains generally consist of IID samples).

Reply:
> This is related, cskip doesn't work with weights, and mcse needs to be passed like this to make sure it works either with or without weights.

That's why I asked you to fix it 🙂 We absolutely want to keep the current behaviour since otherwise in realistic scenarios, such as e.g. the Dirichlet process example IIRC, summary statistics will just be missing.

> naive_se was removed because there's not really a way to define the naive standard error for a weighted chain (since weighted chains generally consist of IID samples). Is there an actual use case for the naive standard errors, given they're inconsistent?

We should discuss this issue separately but not include such changes in this PR. Ie we should keep it for the unweighted/uniformly weighted case.
a

Reply:
> > This is related, cskip doesn't work with weights, and mcse needs to be passed like this to make sure it works either with or without weights.
> 
> That's why I asked you to fix it

Sorry if I was a bit unclear. What I meant is that _`cskip`_ doesn't work with weights, because `cskip` only takes one argument, not that it's impossible to skip missing weights. The most recent PR fixes this by skipping missing observations without using `cskip`.

Reply:
> We should discuss this issue separately but not include such changes in this PR. Ie we should keep it for the unweighted/uniformly weighted case.

I created another issue where this can be discussed, #351 , if you'd like. If you'd like I can create a different PR removing `naive_se` if you don't want to combine them.